### PR TITLE
MGDOBR-61: Add authentication to user API

### PR DESCRIPTION
**Pull request by r00ta**
[MGDOBR-61](https://issues.redhat.com/browse/MGDOBR-61) 

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config)
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [x] All new functionality is tested and uses org.assertj.core.api.Assertions for Assertions
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

File: runner/src/test/resources/application.properties [hunk 1]
```diff
@@ -6,6 +6,4 @@ kafka.security.protocol=PLAINTEXT
 kafka.sasl.jaas.config=none
 
 # SSO
```
Comment (robpblake) [inline runner/src/test/resources/application.properties:8]: Can we set environment variable placeholders here that can be set on the `Deployment` resource definition e.g:

```
quarkus.oidc.auth-server-url=${EVENT_BRIDGE_AUTH_SERVER_URL:some-default}
```

**Comment by robpblake** [comment]
Few minor comments from me @r00ta.

**Review by robpblake — COMMENTED** [review]
🤷‍♂️  I have no idea how I managed that :-) Can you see them now @r00ta

File: runner/src/test/resources/application.properties (hunk 1 above)
Comment (r00ta) [reply runner/src/test/resources/application.properties:8]: @robpblake it is already like this. This is the `application.properties` of the test package

Comment (r00ta) [reply runner/src/test/resources/application.properties:8]: See https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/manager/src/main/resources/application.properties

Comment (robpblake) [reply runner/src/test/resources/application.properties:8]: @r00ta I don't think I should review PRs in GitHub IDE any more 😆

**Review by danielezonca — APPROVED** [review]

**Review by robpblake — APPROVED** [review]

**Pull request merged**